src/syntax_tree/fol.rs

```diff
@@ -918,7 +936,8 @@ impl UserGuide {
         let mut result = IndexSet::new();
         for entry in &self.entries {
             if let UserGuideEntry::PlaceholderDeclaration(p) = entry {
-                result.insert(p.clone());
+                let fc = FunctionConstant::from(p.clone());
+                result.insert(fc);
```

Comment:
Should the function placeholder return placeholder declarations instead of doing the conversion itself?